dragonfly/block/block.go

```diff
@@ -141,3 +143,30 @@ type transparent struct{}
 func (transparent) LightDiffusionLevel() uint8 {
 	return 0
 }
+
+// GravityAffected represents blocks affected by gravity.
+type GravityAffected interface {
+	// CanSolidify returns whether the falling block can return back to a normal block without being on the ground.
+	CanSolidify(pos world.BlockPos, w *world.World) bool
```

Comment:
Is there any circumstance under which someone might want this to return true? I think it might be more useful to simply have a `GravityAffected` method so that different variants of a block could return different values.